clippy_lints/src/declare_clippy_lint.rs

```diff
@@ -0,0 +1,173 @@
+
+        }
+    };
+    (
+        $(#[doc = $lit:literal])*
+        #[clippy::version = $version:literal]
+        pub $lint_name:ident,
+        perf,
+        $desc:literal
+    ) => {
+        declare_clippy_lint! {@
+            $(#[doc = $lit])*
+            pub $lint_name, Warn, crate::LintCategory::Perf,$desc,
+                        declare_clippy_lint!(__version = $version), $version
+        }
+    };
+    (
+        $(#[doc = $lit:literal])*
+        #[clippy::version = $version:literal]
+        pub $lint_name:ident,
+        complexity,
+        $desc:literal
+    ) => {
+        declare_clippy_lint! {@
+            $(#[doc = $lit])*
+                pub $lint_name, Warn, crate::LintCategory::Complexity,$desc,
+                declare_clippy_lint!(__version = $version), $version
+        }
+    };
+    (
+        $(#[doc = $lit:literal])*
+        #[clippy::version = $version:literal]
+        pub $lint_name:ident,
+        suspicious,
+        $desc:literal
+    ) => {
+        declare_clippy_lint! {@
+            $(#[doc = $lit])*
+                pub $lint_name, Warn, crate::LintCategory::Suspicious,$desc,
+                declare_clippy_lint!(__version = $version), $version
+        }
+    };
+    (
+        $(#[doc = $lit:literal])*
+        #[clippy::version = $version:literal]
+        pub $lint_name:ident,
+        nursery,
+        $desc:literal
+    ) => {
+        declare_clippy_lint! {@
+            $(#[doc = $lit])*
+            pub $lint_name,Allow, crate::LintCategory::Nursery,$desc,
+                        declare_clippy_lint!(__version = $version), $version
+        }
+    };
+    (
+        $(#[doc = $lit:literal])*
+        #[clippy::version = $version:literal]
+        pub $lint_name:ident,
+        pedantic,
+        $desc:literal
+    ) => {
+        declare_clippy_lint! {@
+            $(#[doc = $lit])*
+            pub $lint_name,Allow, crate::LintCategory::Pedantic,$desc,
+                        declare_clippy_lint!(__version = $version), $version
+        }
+    };
+    (
+        $(#[doc = $lit:literal])*
+        #[clippy::version = $version:literal]
+        pub $lint_name:ident,
+        cargo,
+        $desc:literal
+    ) => {
+        declare_clippy_lint! {@
+            $(#[doc = $lit])*
+            pub $lint_name,Allow, crate::LintCategory::Cargo,$desc,
+                        declare_clippy_lint!(__version = $version), $version
+        }
+    };
+
+    (
+        $(#[doc = $lit:literal])*
+        $(#[clippy::version = $version:literal])?
+        pub $lint_name:ident,
+        internal,
+        $desc:literal
+    ) => {
+        declare_clippy_lint! {@
+            $(#[doc = $lit])*
+            pub $lint_name,Allow, crate::LintCategory::Internal,$desc,
+            declare_clippy_lint!(__version = $($version)?), "0.0.0"
+        }
+    };
+
+    // VERSION HANDLING
+    (__version = ) => {
+        None
+    };
+    (__version = $version:literal) => {
+        Some($version)
+    };
```

Comment:
I don't think we have an internal lints that have a version, so we should be able to remove the `__version` arms and pass `Some`/`None` directly